db/cache/charts.go

```diff
@@ -826,10 +843,11 @@ var chartMakers = map[string]ChartMaker{
 }
 
 // Chart will return a JSON-encoded chartResponse of the provided type
-// and ZoomLevel.
-func (charts *ChartData) Chart(chartID, zoomString string) ([]byte, error) {
-	zoom := ParseZoom(zoomString)
-	cache, found, cacheID := charts.getCache(chartID, zoom)
+// and BinLevel.
+func (charts *ChartData) Chart(chartID, binString, axisString string) ([]byte, error) {
+	bin := ParseBin(binString)
+	axis := ParseAxis(axisString)
+	cache, found, cacheID := charts.getCache(chartID, bin, axis)
```

Comment:
`TimeAxis` value has major effect only when `DayBin` is set otherwise it doesn't get used.
I also cant see where `WindowBin` is used in the code unless one manually edits the url params to return the "window" value.

Reply:
While `WindowBin` is not used explicitly in `ticketPriceChart` and others window-binned `ChartMaker`s, it is still part of the `cacheKey` for those charts. 

The `axis` is not used for any bin size except `DayBin`. So what happens when you get two requests, one for `/api/chart/hashrate?bin=block&axis=time`, and then another request to `/api/chart/hashrate?bin=block&axis=height`? The same exact data is returned for both, but since the `cacheKey` is different, you are encoding the data twice, unneccesarily. 

Instead of doing that, I'd recommend one of two things

1. When the bin is not `DayBin`, set the axis to some common value, maybe some new value like `NoAxis = ""`. 
2. Find a way to only include the axis in the `cacheKey` when the bin is `DayBin`.